### ISSUE TITLE
Incompatibility with numpy v2?

### ISSUE THREAD

**Issue by NnktYoshioka** [qulacs/qulacs#654]
When I use the newest numpy (v2.0.1) and create a DenseMatrix object it raises a Segmentation fault.
Is this due to pybind version? I have not tested other functions exclusively, but at least some functions such as QuantumState.load did not raise any error.

The minimum example can be reproduced under qulacs==0.6.4.1 and numpy==2.0.1 as follows:

```
from qulacs.gate import DenseMatrix
import numpy as np

pdamp = 0.01

sp = np.array([
        [0, 1],
        [0, 0]
    ])
    
kraus_mat_list = [np.diag([1, np.sqrt(1 - pdamp)]), np.sqrt(pdamp) * sp]
DenseMatrix(0, kraus_mat_list[0])
```

**Comment by KowerKoint** [comment]
Sorry for late reply.
NumPy 2 is currently incompatible.
I'll consider upgrading pybind11 because latest pybind11 supports it.

**Comment by KowerKoint** [comment]
solved by #655
